service/controller/v2/resource/instance/create.go

```diff
@@ -15,11 +18,91 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("EnsureCreated called for cluster ID '%s'", key.ClusterID(customObject)))
+	// Find the next instance ID and instance name we want to trigger the update
+	// for. Instance names look something like the following example.
+	//
+	//     0gjpt-worker-000004
+	//
+	// Instance IDs are simple non negative integers.
+	//
+	//     4
+	//
+	var instanceID string
+	var instanceName string
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "looking for the next instance to be updated")
 
-	// TODO list all instances
-	// TODO find the first instance not having the latest scale set model applied
-	// TODO trigger update for found instance
+		c, err := r.getVMsClient()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		g := key.ResourceGroupName(customObject)
+		s := key.WorkerVMSSName(customObject)
+		result, err := c.List(ctx, g, s, "", "", "")
+		if IsScaleSetNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the scale set")
+			resourcecanceledcontext.SetCanceled(ctx)
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, v := range result.Values() {
+			if *v.LatestModelApplied {
+				continue
+			}
+
+			instanceID = *v.InstanceID
+			instanceName = key.ToInstanceName(customObject, *v.InstanceID)
+
+			if !key.IsFinalProvisioningState(*v.ProvisioningState) {
+				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("instance '%s' is in state '%s'", instanceName, *v.ProvisioningState))
+				resourcecanceledcontext.SetCanceled(ctx)
+				r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
+				return nil
+			}
+
+			break
+		}
+
+		if instanceID == "" {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "no instance ID found that needs to be updated")
+			resourcecanceledcontext.SetCanceled(ctx)
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
+			return nil
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found instance '%s' has to be updated", instanceName))
```

Comment:
I think this log message is redundant.

Reply:
Because we'll have logs like:

```
found instance '%s' has to be updated
ensuring instance '%s' to be updated
```

It looks like the same message in other words. But I guess you want to leave it like that. Do it if it makes you happy.

Reply:
I would like to keep this because it expresses different semantics. It reflects we found the instance and then ensuring the instance we found is updated. These are different things.